data/list/perm.lean

```diff
@@ -429,6 +429,28 @@ theorem subperm_cons (a : α) {l₁ l₂ : list α} : a::l₁ <+~ a::l₂ ↔ l
   { exact ⟨u, perm_cons_inv p, s'⟩ }
 end, λ ⟨l, p, s⟩, ⟨a::l, skip a p, s.cons2 _ _ _⟩⟩
 
+theorem subperm_cons_left {a : α} {l₁ l₂ : list α} (d₁ : nodup l₁) (h₁ : a ∉ l₁) (h₂ : a ∈ l₂)
+ (s : l₁ <+~ l₂) : a :: l₁ <+~ l₂ :=
```

Comment:
I would prefer the name `cons_subperm_of_mem` for this. It doesn't fit the pattern of similarly named theorems around it like `subperm_cons` and `subperm_app_left` that are biconditionals.